apps/explorer/src/hooks/useGetObject.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type GetObjectDataResponse } from '@mysten/sui.js';
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+
+import { useRpc } from './useRpc';
+
+export function useGetObject(
+    objectId: string
+): UseQueryResult<GetObjectDataResponse, unknown> {
+    const rpc = useRpc();
+    const response = useQuery(
+        ['object', objectId],
+        async () => {
+            return rpc.getObject(objectId);
+        },
+        { enabled: !!objectId, staleTime: Infinity }
```

Comment:
Leave stale time at the default. We can make the hook take additional options in the future if we really do want infinite stale time.